src/tox_lsr/test_scripts/runcollection.sh

```diff
@@ -12,19 +13,26 @@ SCRIPTDIR=$(readlink -f "$(dirname "$0")")
 # Collection commands that are run when `tox -e collection`:
 role=$(basename "${TOPDIR}")
 STABLE_TAG=${2:-master}
-LSR_ROLE2COLL_NAMESPACE="${LSR_ROLE2COLL_NAMESPACE:-fedora}"
-LSR_ROLE2COLL_NAME="${LSR_ROLE2COLL_NAME:-linux_system_roles}"
-cd "$LSR_TOX_ENV_DIR"
+export LSR_ROLE2COLL_NAMESPACE="${LSR_ROLE2COLL_NAMESPACE:-fedora}"
+export LSR_ROLE2COLL_NAME="${LSR_ROLE2COLL_NAME:-linux_system_roles}"
+
+# Since .tox is in .gitignore, ansible-test is skipped
+# if the collection path is in LSR_TOX_ENV_DIR.
+# Using a temp dir outside of .tox.
+export MY_LSR_TOX_ENV_DIR=$( mktemp -d -t tox-XXXXXXXX )
+cd "$MY_LSR_TOX_ENV_DIR"
 testlist="yamllint,black,flake8,shellcheck"
 # py38 - pyunit testing is not yet supported
 #testlist="${testlist},py38"
 
 automaintenancerepo=https://raw.githubusercontent.com/linux-system-roles/auto-maintenance/
 curl -s -L -o lsr_role2collection.py "${automaintenancerepo}${STABLE_TAG}"/lsr_role2collection.py
 
-python lsr_role2collection.py --src-path "$TOPDIR/.." --dest-path "$LSR_TOX_ENV_DIR" --role "$role" \
+python lsr_role2collection.py --src-path "$TOPDIR/.." --dest-path "$MY_LSR_TOX_ENV_DIR" --role "$role" \
   --namespace "${LSR_ROLE2COLL_NAMESPACE}" --collection "${LSR_ROLE2COLL_NAME}" \
-  > "$LSR_TOX_ENV_DIR"/collection.out 2>&1
+  > "$MY_LSR_TOX_ENV_DIR"/collection.out 2>&1
+
+trap "rm -rf ${MY_LSR_TOX_ENV_DIR}" 0
```

Comment:
The trap should go just after the `mktemp` so that if anything unexpected happens, the dir will be removed

Reply:
:p